src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/ClientsApiController.java

```diff
@@ -527,10 +544,70 @@ private ClientType getClientType(String encodedId) {
         List<ServiceClient> serviceClients = null;
         try {
             serviceClients = serviceClientConverter.
-                    convertAccessRightHolderDtos(accessRightService.getAccessRightHoldersByClient(clientId));
+                    convertServiceClientDtos(serviceClientService.getServiceClientsByClient(clientId));
         } catch (ClientNotFoundException e) {
             throw new ResourceNotFoundException(e);
         }
         return new ResponseEntity<>(serviceClients, HttpStatus.OK);
     }
+
+    @Override
+    @PreAuthorize("hasAuthority('VIEW_CLIENT_ACL_SUBJECTS')")
+    public ResponseEntity<ServiceClient> getServiceClient(String id, String scId) {
```

Comment:
FYI old implementation gives frontend also boolean `has_services`

```
    render_json({
      :acl_subjects => read_acl_subjects(client),
      :has_services => has_services
    })

```
Frontend has some logic that uses that, it disables the wizard, or parts of wizard based on that info. Probably needs to be taken into account when doing the frontend ticket for add service client access rights?

Reply:
This could also be handled in the frontend as it's only empty array -check. What do you think?

Reply:
Not sure if it is just empty array -check (doesn't it mean a call to `/clients/{id}/service-descriptions` and then iterating ServiceDescriptions -> Services?) - but in any case yes, it is probably better to handle it in the frontend than add some extra baggage to this endpoint.